acousticbrainz/views/data.py

```diff
@@ -44,21 +46,25 @@ def view_high_level(mbid):
 
 @data_bp.route("/<uuid:mbid>", methods=["GET"])
 def summary(mbid):
-    lowlevel, highlevel, genres, moods, other = get_summary_data(mbid)
+    summary = get_summary_data(mbid)
+    
+    info = _get_track_info(mbid, summary['lowlevel']['metadata'] if summary['lowlevel'] else None)
 
-    info = _get_track_info(mbid, lowlevel['metadata'])
+    tomahawk_url = _get_tomahawk_url(info)
 
-    # Tomahawk player stuff
-    if not ('artist' in lowlevel['metadata']['tags'] and 'title' in lowlevel['metadata']['tags']):
-        tomahawk_url = None
-    else:
-        tomahawk_url = "http://toma.hk/embed.php?artist={artist}&title={title}".format(
-            artist=quote_plus(info['artist'].encode("UTF-8")),
-            title=quote_plus(info['title'].encode("UTF-8")))
+    if info:
+        return render_template("data/summary.html", summary=summary, mbid=mbid, info=info, tomahawk_url=tomahawk_url)
```

Comment:
Also, you can replace second `tomahawk_url` on this line with `_get_tomahawk_url(info)` and remove assignment above.